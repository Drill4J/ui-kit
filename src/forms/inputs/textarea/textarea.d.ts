import { TextareaProps } from './textarea-types';
export declare const Textarea: {
    (props: TextareaProps): JSX.Element;
    displayName: string;
    __docgenInfo: {
        description: string;
        displayName: string;
        props: {
            className: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            value: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            onChange: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            disabled: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            error: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
        };
    };
};
