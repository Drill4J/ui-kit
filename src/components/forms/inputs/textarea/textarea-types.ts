export interface TextareaProps {
  className?: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  disabled?: boolean;
  error?: boolean;
}
