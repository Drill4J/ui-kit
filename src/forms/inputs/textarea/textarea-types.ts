export interface TextareaProps {
  className?: string;
  value: string;
  onChange: (event: React.SyntheticEvent<HTMLTextAreaElement>) => void;
}
