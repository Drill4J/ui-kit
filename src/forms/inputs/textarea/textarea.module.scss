@import '../../../style-guide/index';

.textarea {
  box-sizing: border-box;
  width: 100%;
  height: 80px;
  border-radius: 4px;
  border: 1px solid $loblolly-color;
  resize: none;
  font-family: $regular-font;
  font-size: 14px;
  outline: none;
  line-height: 40px;
  padding: 0 17px;
  color: map-get($color-palette, dark-blue);

  &::placeholder {
    opacity: 0.5;
  }

  &--error {
    border: 1px solid map-get($color-palette, red100);
  }
}
