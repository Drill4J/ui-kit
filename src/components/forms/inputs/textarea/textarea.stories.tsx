import React from 'react';
import { ComponentMeta } from '@storybook/react';
import 'twin.macro';

import { Textarea } from './textarea';

export default {
  title: 'Textarea',
  component: Textarea,
} as ComponentMeta<typeof Textarea>;

const Template = (args) => <Textarea {...args} />;

export const Default = Template.bind({});
Default.args = { disabled: false };

export const Disabled = Template.bind({});
Disabled.args = { disabled: true };
