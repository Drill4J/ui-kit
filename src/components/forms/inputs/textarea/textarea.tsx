import tw, { styled } from 'twin.macro';

import { TextareaProps } from './textarea-types';

export const Textarea = styled.textarea<TextareaProps>`
  ${tw`box-border w-full h-full rounded font-regular text-14 leading-20 border border-monochrome-medium-tint outline-none resize-none
    focus:border-monochrome-black
    placeholder:text-monochrome-default
  `}
  padding: 10px 16px;

  ${({ disabled, error }) => [
    disabled && tw`border-monochrome-medium-tint bg-monochrome-light-tint text-monochrome-default`,
    error && tw`border-red-default`,
  ]}
`;
