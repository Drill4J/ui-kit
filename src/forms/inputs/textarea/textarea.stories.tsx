import React from 'react';
import { storiesOf } from '@storybook/react';

import { Textarea } from './textarea';

storiesOf('Textarea', module).add('Textarea', () => (
  <Textarea value="Textarea component" onChange={() => {}} />
));
