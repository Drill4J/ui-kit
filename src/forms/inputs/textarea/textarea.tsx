import { BEM, textarea as bemTextarea } from '@redneckz/react-bem-helper';

import { TextareaProps } from './textarea-types';
import { INPUT_PROPS } from '../input/input-props';

import styles from './textarea.module.scss';

const textarea = BEM(styles);

const INPUT_PROPS_OBJ = Object.assign({}, ...INPUT_PROPS.map((key) => ({ [key]: undefined })));

export const Textarea = (props: TextareaProps) => <TextareaElement {...props} />;

const TextareaElement = textarea.input(
  bemTextarea({
    ...INPUT_PROPS_OBJ,
    value: '',
  } as {}),
);
