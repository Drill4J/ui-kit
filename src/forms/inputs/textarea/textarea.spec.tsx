import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Textarea } from './textarea';

describe('Textarea', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Textarea value="Textarea component" onChange={() => {}} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with initial state', () => {
    const mockFn = jest.fn();
    const props = {
      value: 'textarea',
      onChange: mockFn,
    };
    const wrapper = mount(<Textarea {...props} />);
    wrapper.find('textarea').simulate('change');
    expect(mockFn).toHaveBeenCalled();
  });
});
