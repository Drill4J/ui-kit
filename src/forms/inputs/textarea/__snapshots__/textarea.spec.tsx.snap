// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should match snapshot 1`] = `
<textarea
  className="textarea"
  data-modifiers=""
  onChange={[Function]}
  value="Textarea component"
/>
`;
